### DISCUSSION TITLE
Do RS-485 PCBs that don't have an RTS pin support RDM?

### DISCUSSION THREAD

**Discussion by yliniemi** [someweisguy/esp_dmx#109] [General]
I have a bunch of these boards
https://www.bizkit.ru/wp-content/uploads/2019/01/RS485_interface_module_Arduino.png
They have some extra protection against power surges and what not. But they also lack the pin to select whether they're transmitting or receiving. They somehow know automatically what the direction is. No RE or DE pins on this board.

When I use this to either just transmit or just receive I assume the circuitry can figure out which one it is doing. But what I'm interested is RDM. If I understand correctly with RDM the PCB first has to transmit an RDM query and after that receive the answer through the same channel. What my question is that can these auto sensing PCBs handle changing their state fast enough to support RDM?

I can live without RDM. Not many cheap fixtures that I come across support it anyways. But I was just curious if I might run into issues with these autosensing RS-485 PCBs when trying to implement RDM.

**Comment by someweisguy** [comment]
I have used these RS485 modules and have had difficulties with them in the past so I try to avoid using them. 

> If I understand correctly with RDM the PCB first has to transmit an RDM query and after that receive the answer through the same channel.

This is correct! The RDM controller sends a request to a specific UID and the device with that UID sends a response.

>They somehow know automatically what the direction is. No RE or DE pins on this board.

As far as I know, these RS485 modules will auto-trigger the RTS pin whenever a 1 is written to the TX pin of the RS485 module. This means that the RDM controller will put the DMX line in a high-impedance state every time a 0 is written. Technically this is not allowed by the RDM standard. Section 3.1.3 of the RDM standard says `Controller drivers shall drive the line (not enter a high impedance state) throughout the entire packet.` I think you would be able to send RDM commands to some devices but I wouldn't bet on being able to achieve reliable RDM control with that RS485 module. 

Hope this helps! Let me know if you have other questions!

**Reply by yliniemi** [reply]
Thanks! This helps immensely.

I didn't quite understand the high impedance thingy you talk about. Perhaps I'd have to look at some drawings about the RDM protocol to understand what you're talking about. But perhaps I can just bypass having to understand that.

Is RDM unreliable with this board? I will not support RDM then.
You said you've had issues with this board. What kind? Something major? I can live without RDM support but if regular DMX transmitting or receiving is problematic I will choose the board you recommend.

I'm mainly going to be doing transmitting DMX. If this board has problems receiving DMX I can just restrict them to send only. But if they have issues transmitting DMX they are going to the trash right away.

**Reply by someweisguy** [reply]
Apologies for the delay, I tried to find some notes from when I last tested with this device. I remembered that I had issues with the device, but I couldn't recall exactly what went wrong. I would have tested with it again, but unfortunately I recently moved across the country and have since thrown away all of these modules that I used to own in the move.

After taking more time to think about it, it occurred to me that **this RS485 module may not work as a DMX transmitter**. The reason for this is because the module only controls the RS485 bus when a 1 is transmitted over the line. This means that sending a DMX break, a 0, is impossible. This would be the case with this RS485 module for every DMX library - not just this one. Receiving DMX will work. Since sending DMX won't work, RDM will effectively not work either.

Hopefully this is helpful and makes sense. If there is a wiring diagram that comes with this RS485 module, I would be able to tell for sure about DMX transmitting not working. But I am 90% sure that this was the issue I ran into.
